src/workerd/api/streams/standard.c++

```diff
@@ -908,8 +908,10 @@ void ReadableImpl<Self>::doCancel(
 
   auto onSuccess = JSG_VISITABLE_LAMBDA((this, self = self.addRef()), (self), (jsg::Lock& js) {
     doClose(js);
-    KJ_IF_MAYBE(pendingCancel, maybePendingCancel) {
-      maybeResolvePromise(js, pendingCancel->fulfiller);
+    KJ_IF_SOME(pendingCancel, maybePendingCancel) {
+      maybeResolvePromise(js, pendingCancel.fulfiller);
+    } else {
+      // Else block to avert dangling else compiler warning.
```

Comment:
huh, what is the compiler warning about when the `else` isn't here? Does KJ_IF_SOME generate an empty else?

Reply:
https://github.com/capnproto/capnproto/blob/761aeb17563a59f43b3fe9bae93df83c6bd57d06/c%2B%2B/src/kj/common.h#L1353

`KJ_IF_SOME` has a dangling if suppressions, but the compiler still complains in a few places. Complaints typically come with nested `KJ_IF_SOME` blocks, but also occur with a few others.

Reply:
Can we fix `KJ_IF_SOME` for those cases?

Reply:
I don't think we have a fix for now, or at least one we like, @harrishancock might have thoughts on this. I've come across maybe 7-8 places that it is an issue. For now, it seems reasonable to include the small intervention and document it, so it is a quick clean-up in future.

The problem is caused by `KJ_IF_SOME` wanting to support `KJ_IF_SOME(..) {...} else {...}`, when two are nested there are two `if` statements without else blocks and that is where the ambiguity comes from.

One approach might be to replace the first `if` with a `for (bool done303 = false, ...; !done303; done303 = true)` that includes a temporary flag that only iterates the loop once; but this adds overhead and debugger visibility of the temporary. An optimized build may be able to avoid the temporary. Not tried.

Reply:
> We might be able replace an `if` for a `for`, but that might also have drawbacks. Not tried.

Ignore this, I don't see a way to make that work and support nesting.